### DISCUSSION TITLE
Apollo Initialization

### DISCUSSION THREAD

**Discussion by yordis** [vercel/next.js#18431] [Help] [answered]
Is there any reason why I can't have a singleton for the Apollo client in SSR?

Follow up question to this comment: https://github.com/vercel/next.js/blob/44fe971e6b3ba240ecd8f2d2d92025d831bc609d/examples/with-apollo/lib/apolloClient.js#L38

**Comment by yordis** [comment]
@lfades  maybe you know the answer?

**Comment by lfades** [comment, marked as answer]
SSR happens in Node.js, you need to have a single client for that because the same client can't be send to the browser, its not possible or encouraged. While using Apollo, in SSR we generate static HTML and a JSON for the data using the Apollo Client, then in the browser that initial static HTML and JSON is used for the initial render, and then another Apollo Client is created in the browser and populated with the initial data (from the JSON generated in SSR with Node.js).

If you also have this question: "But why can't I use the same Apollo Client in my Node.js environment?". The answer is because if you do so, you may end up merging data between the visitors of your app, and that will be very bad.

**Reply by yordis** [reply]
> The answer is because if you do so, you may end up merging data between the visitors of your app, and that will be very bad.

Jeez, I completely forgot. That totally makes sense.

Thanks a lot!
